doc/source/development/test_writing.rst

```diff
@@ -0,0 +1,152 @@
+
+    - tests.frame.test_reductions
+    - tests.series.test_reductions
+    - tests.test_nanops
+
+5. Is your test for an indexing method?
+    This is the most difficult case for deciding where a test belongs, because
+    there are many of these tests, and many of them test more than one method
+    (e.g. both ``Series.__getitem__`` and ``Series.loc.__getitem__``)
+
+    A) Is the test specifically testing an Index method (e.g. ``Index.get_loc``, ``Index.get_indexer``)?
+    This test likely belongs in one of:
+
+        - tests.indexes.test_indexing
+        - tests.indexes.fooindex.test_indexing
+
+    Within that files there should be a method-specific test class e.g. ``TestGetLoc``.
+
+    In most cases, neither ``Series`` nor ``DataFrame`` objects should be needed in these tests.
+
+    B) Is the test for a Series or DataFrame indexing method *other* than ``__getitem__`` or ``__setitem__``, e.g. ``xs``, ``where``, ``take``, ``mask``, ``lookup``, or ``insert``?
+    This test likely belongs in one of:
+
+        - tests.frame.indexing.test_methodname
+        - tests.series.indexing.test_methodname
+
+    C) Is the test for any of ``loc``, ``iloc``, ``at``, or ``iat``?
+    This test likely belongs in one of:
+
+        - tests.indexing.test_loc
+        - tests.indexing.test_iloc
+        - tests.indexing.test_at
+        - tests.indexing.test_iat
+
+    Within the appropriate file, test classes correspond to either types of indexers (e.g. ``TestLocBooleanMask``) or major use cases (e.g. ``TestLocSetitemWithExpansion``).
+
+    See the note in section D) about tests that test multiple indexing methods.
+
+    D) Is the test for ``Series.__getitem__``, ``Series.__setitem__``, ``DataFrame.__getitem__``, or ``DataFrame.__setitem__``?
+    This test likely belongs in one of:
+
+        - tests.series.test_getitem
+        - tests.series.test_setitem
+        - tests.frame.test_getitem
+        - tests.frame.test_setitem
+
+    If many cases such a test may test multiple similar methods, e.g.
+
+    .. code-block:: python
+
+         import pandas as pd
+         import pandas._testing as tm
+
+         def test_getitem_listlike_of_ints():
+             ser = pd.Series(range(5))
+
+             result = ser[[3, 4]]
+             expected = pd.Series([2, 3])
+             tm.assert_series_equal(result, expected)
+
+             result = ser.loc[[3, 4]]
+             tm.assert_series_equal(result, expected)
+
+    In cases like this, the test location should be based on the *underlying* method being tested.  Or in the case of a test for a bugfix, the location of the actual bug.  So in this example, we know that ``Series.__getitem__`` calls ``Series.loc.__getitem__``, so this is *really* a test for ``loc.__getitem__``.  So this test belongs in ``tests.indexing.test_loc``
+
+6. Is your test for a DataFrame or Series method?
+    A) Is the method a plotting method?
```

Comment:
When starting a sub-item (the A) list item), there needs to be a blank line preceding it to make the separation with the previous (parent) item clear